core/json/json_path.rs

```diff
@@ -0,0 +1,83 @@
+use pest_derive::Parser;
+
+#[derive(Parser)]
+#[grammar_inline = r#"
+array_locator = @{ "[" ~ array_index ~ "]" }
```

Comment:
I had to copy a large part of grammar from `core/json/de.rs`. Perhaps I could extract the grammar from there into a constant and add additional rules for parsing the path? However, this would leave us with a few unused rules.

The other way to DRY it out would be to extract only the common rules, but I feel like it's brittle and very unhandy if we ever have to patch the JSON grammar.